src/interface_adapter/room_settings/OpenRoomSettingsController.java

```diff
@@ -0,0 +1,29 @@
+package interface_adapter.room_settings;
+
+import entities.auth.User;
+import entities.rooms.Room;
+import interface_adapter.ViewManagerModel;
+import view.RoomSettingsView;
+
+public class OpenRoomSettingsController {
+  // We diverge from Clean Architecture slightly here as is recommended on piazza:
+  // https://piazza.com/class/lm84a7mo56k5wy/post/1231
+  // In particular, we only need to modify the view model, and not use any data access objects or
+  // other entities
+
+  RoomSettingsViewModel roomSettingsViewModel;
+  ViewManagerModel viewManagerModel;
```

Comment:
This is not ok; please find a way to put this in the presenter. Taking shortcuts like this may seem reasonable since we are nearing the end of the project and will have to change less code, but we should still adhere to CA when it makes sense and is possible.

Reply:
Is this even a controller if this is all the class does? There must be a better way to do this.

Reply:
Switch view was created to avoid this.